core/src/main/scala/org/bitcoins/core/crypto/bip32/BIP32Path.scala

```diff
@@ -0,0 +1,88 @@
+package org.bitcoins.core.crypto.bip32
+
+import org.bitcoins.core.crypto.ExtKey
+import org.bitcoins.core.number.UInt32
+
+abstract class BIP32Path {
+  def children: Vector[BIP32Child]
```

Comment:
So I think i would prefer to name this `path` rather than `children`. From my understanding, `Vector[BIP32Child]` does _not_ contain only children. 

In the case of `BIP44Coin`, it actually appears that it is the opposite. It appears that it contains itself, and all of it's ancestors? 

https://github.com/bitcoin-s/bitcoin-s-core/pull/379/files#diff-eaf823968efd2e0e07cc92b04ae905cbR10